playbooks/deploy-external-ceph.yml

```diff
@@ -0,0 +1,253 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Pre-requisites for deploying external ceph
+  hosts: localhost
+  vars:
+    cifmw_ceph_dir: "{{ ansible_user_dir }}/ci-framework-data/manifests/ceph"
+    cifmw_ceph_node_num: "{{ cifmw_ceph_node_num | default(3) }}"
+  tasks:
+    - name: Prepare inventory for ceph nodes
+      block:
+
+        - name: Add hosts to ceph group
+          ansible.builtin.add_host:
+            name: "{{ item }}"
+            groups: ceph
+          loop: "{{ groups['computes'][-cifmw_ceph_node_num:] }}"  
+          when: groups['computes'] | length > {{ cifmw_ceph_node_num }}
+          run_once: true
+          
+        - name: Remove ceph nodes from the compute group
+          ansible.builtin.add_host:
+            name: "{{ item }}"
+            groups: computes
+            state: absent
+          loop: "{{ groups['ceph'] }}"
+          when: item in groups['computes']
+          
+    - name: Gather IP of the ceph nodes
+      block:
+        - name: Run command to get IP
+          ansible.builtin.shell:
+            cmd: "ip a | grep 192.168.122 | awk '{print $2}' | cut -d '/' -f 1"
+          register: ip_result
+          with_items: "{{ groups['ceph'] }}"
+          delegate_to: "{{ item }}"
+
+        - name: Store IPs in a dictionary
+          set_fact:
+            ceph_ips_dict: "{{ ceph_ips_dict | default({}) | combine({item.item: item.stdout_lines[0]}) }}"
+          with_items: "{{ ip_result.results }}"
+
+
+    - name: Create a nodeset for ceph
+      block:
+        - name: Ensure that directory exists
+          ansible.builtin.file:
+            path: "{{ cifmw_ceph_dir }}"
+            state: directory
+
+        - name: Get sample nodeset file from URL
+          get_url:
+            url: "https://raw.githubusercontent.com/openstack-k8s-operators/dataplane-operator/main/config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml"
```

Comment:
This will break after this merges.

  https://github.com/openstack-k8s-operators/dataplane-operator/pull/698